orm/bucket.go

```diff
@@ -51,17 +51,17 @@ type Bucket struct {
 var _ weave.QueryHandler = Bucket{}
 
 type namedIndex struct {
-	Index
```

Comment:
I dont really understand the changes here. 

Why and what is the goal?
For versions or general cleanup?

In any case, it does bring up the issue that we need to consider how versioned data interacts with secondary indexes. 

My gut feel is only to index latest. Older versions only available with direct (Id, version) reference